sktime/_contrib/datasets/_single_problem_loaders.py

```diff
@@ -936,6 +936,9 @@ def load_macroeconomic():
     .. [3] Data Source: Bureau of Labor Statistics, U.S. Department of Labor;
           http://www.bls.gov/data/; accessed December 15, 2009.
     """
+    _check_soft_dependencies("statsmodels")
```

Comment:
Shouldn't we introduce `_check_soft_dependencies("statsmodels")` before every statsmodels import then? In #3445 most (maybe all) of the updated files don't have that.

Reply:
the trick is in the `python_dependencies` tag, which has the value `"statsmodels"`, in those cases.

If that is set, the check is called by the base class (which reads the tag). Single locus of responsibility, for lookup as well as dependency checking.

Reply:
> Is this a statsmodels-specific thing?

No, generally we want ot isolate soft dependencies to inside classes or functions. Isolating statsmodels is a "good thing" imo, see #3445, even if it remains a core dependency.

Reply:
Ok, I understand, thanks for clarification, Franz